File: JavaScript_Courses/JavaScript_Basics/Preparation/05_While_Loop_Exercise/07_Moving.js
function moving(input) {
    let width = Number(input[0]);
    let length = Number(input[1]);
    let heigth = Number(input[2]);

    let spaceAvailable = width * length * heigth;
    let i = 3;

    while (true) {
        let currentCommand = input[i];
        
        if (currentCommand == "Done") {
            console.log(`${spaceAvailable} Cubic meters left.`);
            break;
        }

        let currentSpace = Number(currentCommand);
        spaceAvailable -= currentSpace;

        if (spaceAvailable < 0) {
            console.log(`No more free space! You need ${-spaceAvailable} Cubic meters more.`);
            break;
        }

        i++;
    }
}


// moving([
//     "10",
//     "10",
//     "2",
//     "20",
//     "20",
//     "20",
//     "20",
//     "122",
// ]);

File: Python_Courses/Python_Basics/Python_PB_2025_01/05_While_Loop_Exercise/07_Moving.py
width = int(input())
length = int(input())
height = int(input())

total_space_left = width * length * height

command = input()  # 'Done' or space as a string, e.g. '20'
while command != 'Done':
    space = int(command)  # e.g. int('20') == 20

    total_space_left -= space
    if total_space_left <= 0:
        break

    command = input()

is_space_left = total_space_left >= 0
if not is_space_left:
    print(f'No more free space! You need {-total_space_left} Cubic meters more.')
else:
    print(f'{total_space_left} Cubic meters left.')
